modules/stochastic_tools/src/distributions/WeibullDistribution.C

```diff
@@ -0,0 +1,28 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "WeibullDistribution.h"
+
+template <>
+InputParameters
+validParams<WeibullDistribution>()
+{
+  InputParameters params = validParams<Distribution>();
+  params.addClassDescription("Boost Weibull distribution.");
+  params.addRequiredParam<Real>("shape", "The Weibull shape parameter.");
+  params.addParam<Real>("scale", 1, "The Weibull scale parameter.");
+  return params;
+}
+
+WeibullDistribution::WeibullDistribution(const InputParameters & parameters)
+  : BoostDistribution(parameters)
```

Comment:
You are missing the template parameter here in the call to the base class. This is a common mistake that used to catch us all the time on at least one of the compilers we tested against (maybe the minimum) We've had to fix this over and over. It looks like the rules are changing with C++11 but you might as well be careful here:
https://stackoverflow.com/questions/8887864/template-base-constructor-call-in-member-initialization-list-error